geoh5py/shared/concatenation/drillholes_group_table.py

```diff
@@ -0,0 +1,101 @@
+#
+#  geoh5py is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  geoh5py is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+from abc import ABC
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+if TYPE_CHECKING:
+    from .drillholes_concatenator import DrillholesConcatenator
+
+
+class DrillholesGroupTable(ABC):
+    """
+    A class to store the information of a PropertyGroup.
+
+    :param parent: The parent DrillholesConcatenator.
+    :param name: The name of the PropertyGroup to extract.
+    """
+
+    def __init__(self, parent: DrillholesConcatenator, name: str):
+        if not hasattr(parent, "unique_property_group_names"):
+            raise TypeError("The parent must be a Concatenator instance.")
+
+        self._parent: DrillholesConcatenator = parent
+
+        # ensure name is in parent's unique_property_group_names
+        if name not in parent.unique_property_group_names:
+            raise KeyError(
+                f"The name '{name}' is not in the parent's unique_property_group_names."
+            )
+
+        self._name: str = name
+
+    # get all the property groups in the concatenator
+    def association_by_drillhole(
+        self,
+        names: tuple,
+    ) -> dict:
+        """
+        Based on the first name of the input list,
+        get for every object index and count of all the data in 'names'
+
+        :param names: The names of the data to extract.
+
+        :return: A dictionary with the object uuid and the index of all the data.
+        """
+        # ensure all the names are in the index
+        if not all(name in self._parent.index for name in names):
+            raise KeyError(f"Data '{names}' not found in concatenated data.")
+
+        object_index: dict = {}
+        # todo: is the order of the index always the same?
+        for drillhole in np.sort(self._parent.index[names[0]], order="Start index")[
+            "Object ID"
+        ]:
+            object_index[drillhole] = {}
+            for name in names:
+                if drillhole in self._parent.index[name]["Object ID"]:
+                    object_index[drillhole][name] = list(
+                        self._parent.index[name][
+                            self._parent.index[name]["Object ID"] == drillhole
+                        ][0]
+                    )[:2]
+                else:
+                    object_index[drillhole][name] = [0, 0]
```

Comment:
Make this chunk a method of Concatenator, something like ?

`def get_data_interval(self, drillhole: uuid.UUID, name: str) -> tuple:`